skrf/calibration/calibration.py

```diff
@@ -6472,24 +6472,35 @@ def convert_8term_2_12term(coefs_8term):
     k_first = coefs_8term.get('k first', k)
     k_second = coefs_8term.get('k second', k)
 
-    # taken from eq (36)-(39) in the Roger Marks paper given in the
-    # docstring
-    Elf  = Esr + (Err*gamma_f)/(1. - Edr * gamma_f)
-    Elr = Esf  + (Erf *gamma_r)/(1. - Edf  * gamma_r)
-    Etf  = ((Elf  - Esr)/gamma_f) * k_first
-    Etr = ((Elr - Esf )/gamma_r) * 1./k_second
-
     coefs_12term = {}
-    for l in ['forward directivity','forward source match',
-        'forward reflection tracking','reverse directivity',
-        'reverse reflection tracking','reverse source match',
-        'forward isolation', 'reverse isolation']:
-        coefs_12term[l] = coefs_8term[l].copy()
+
+    if np.allclose(gamma_f, np.zeros_like(gamma_f))\
+            or np.allclose(gamma_r, np.zeros_like(gamma_r)):
```

Comment:
If only one switch term is all zeros then the other is also treated as if it was zero resulting in incorrect result if that's not the case. Division by zero also remains if only some elements in switch terms are zero.

Reply:
Good morning,
I just pushed another suggestion: gamma_r and gamma_f are now checked independently and in the case of being all-zero the reduced equations are taken. That means that none of the coeffs (forward/reverse load match, transmission tracking) will be nan after the conversion from 8- to 12-terms.

Reply:
I don't see any new commits.

Reply:
Sorry, again my SSH agent has removed my SSH key. 
I pushed it now and moreover added tests that cover all cases: Both switch terms are there, none, and either of the two).